pkg/operator/clustermemberremovalcontroller/clustermemberremovalcontroller.go

```diff
@@ -97,10 +101,85 @@ func (c *clusterMemberRemovalController) sync(ctx context.Context, _ factory.Syn
 	if err := c.attemptToRemoveLearningMember(ctx); err != nil {
 		errs = append(errs, err)
 	}
+	if err := c.attemptToScaleDown(ctx); err != nil {
+		errs = append(errs, err)
+	}
 	return kerrors.NewAggregate(errs)
 
 }
 
+// attemptToScaleDown attempts to remove a voting member only once we have identified that
+// a Machine resource is being deleted and a replacement member has been created
+func (c *clusterMemberRemovalController) attemptToScaleDown(ctx context.Context) error {
+	currentVotingMemberIPListSet, err := c.votingMemberIPListSet()
+	if err != nil {
+		return err
+	}
+
+	desiredControlPlaneReplicasCount, err := ceohelpers.ReadDesiredControlPlaneReplicasCount(c.operatorClient)
+	if err != nil {
+		return err
+	}
+	if desiredControlPlaneReplicasCount == 0 {
+		return fmt.Errorf("desired control plane replicas count cannot be empty")
+	}
+	if currentVotingMemberIPListSet.Len() <= desiredControlPlaneReplicasCount {
```

Comment:
how will you handle the case on metal where the cluster first needs to be scaled down and then is scaled back up?

Reply:
> In short: When a Machine is marked for deletion, if the hook is removed, the etcd operator will not replace it. This will allow users to override the mechanism and continue with their replacement in these scenarios.

and you expect these external parties to handle an etcd member removal?  I don't think that's reasonable.

Reply:
> the etcd operator should be able to detect that the Machine object has been removed and remove the member in these scenarios

implemented in https://github.com/openshift/cluster-etcd-operator/commit/6b8f7c8185adfe08618c7e0ca86adf80803a7994

Reply:
Thanks for that link @p0lyn0mial , so my memory was correct on the plan then, question is, is @deads2k happy with the UX I've described in my previous comment

Reply:
I suspect that will end up bumpy on etcd consumers.  I'm willing to merge this to get started, but I suspect we'll have an issue.